components/support/nimbus-fml/fixtures/fe/default_merging.yaml

```diff
@@ -0,0 +1,89 @@
+---
+features:
+  dialog-appearance:
+    description: This is the appearance of the dialog
+    variables:
+      positive:
+        description: This is a positive button
+        type: Button
+        default:
+          {
+            "label": "Ok then",
+            "color": "blue"
+          }
+      negative:
+        description: This is a negative button
+        type: Button
+        default:
+          {
+            "label": "Not this time",
+            "color": "red"
+          }
+      background-color:
+        description: This is the background color
+        type: String
+        default: white
+      player-mapping:
+        description: This is the map of the player type to a button
+        type: Map<PlayerProfile, Button>
+        default: {
+          "child": {},
+          "adult": {}
```

Comment:
A good test would test that

 * A scalar feature variable could be replaced.
 * An object feature variable with a scalar field being replace
 * A map feature variable with a scalar value being replace.
 * An object feature variable with a map with a scalar value being replaced.
 
etc. etc. 

Think of test cases to break your code.